src/screens/WalletType.js

```diff
@@ -0,0 +1,216 @@
+        if (arg.success) {
+          // We wait 2 seconds to update the message on the screen
+          // so the user don't see the screen updating fast
+          setTimeout(() => {
+            this.setState({ waitAction: true, errorMessage: '' }, () => {
+              ledger.getPublicKeyData();
+            })
+          }, 2000);
+        } else {
+          if (!this.state.hardware) {
+            // User clicked 'Back' button
+            this.trialNumber = 0;
+            return;
+          }
+
+          if (this.trialNumber < this.trialLimit) {
+            setTimeout(() => {
+              this.trialNumber += 1;
+              ledger.getVersion();
+            }, this.trialInterval);
+          } else {
+            // Error
+            this.setState({ errorMessage: arg.error });
+          }
+        }
+      });
+
+      IPC_RENDERER.on("ledger:publicKeyData", (event, arg) => {
+        if (!this.state.hardware) {
+          // User clicked 'Back' button
+          return;
+        }
+
+        if (arg.success) {
+          const data = arg.response;
+          const uncompressedPubkey = data.slice(0, 65);
+          const compressedPubkey = hathorLib.wallet.toPubkeyCompressed(uncompressedPubkey);
+          const chainCode = Buffer.from(data.slice(65, 97));
+          const fingerprint = Buffer.from(data.slice(97, 101));
+          const xpub = hathorLib.wallet.xpubFromData(compressedPubkey, chainCode, fingerprint);
+          // First we clean what can still be there of a last wallet
+          wallet.cleanWallet();
+          wallet.startHardwareWallet(xpub);
+          hathorLib.wallet.markBackupAsDone();
+          this.props.history.push('/wallet/');
+        } else {
+          // Error
+          this.setState({ errorMessage: arg.error.message });
+        }
+      });
+    }
+  }
+
+  componentWillUnmount() {
+    if (IPC_RENDERER) {
+      // Remove listeners
+      IPC_RENDERER.removeAllListeners("ledger:version");
+      IPC_RENDERER.removeAllListeners("ledger:publicKeyData");
+    }
+  }
+
+  /**
+   * Go to signin screen
+   */
+  goToSoftwareWallet = () => {
+    hathorLib.wallet.setWalletType('software');
+    this.props.history.push('/software_warning/');
+  }
+
+  /**
+   * Go to the hardware wallet screen
+   */
+  goToHardwareWallet = () => {
+    hathorLib.wallet.setWalletType('hardware');
+    this.setState({ hardware: true }, () => {
+      this.trialNumber = 1;
+      ledger.getVersion();
+    });
+  }
+
+  /*
+   * Try getting user approval on Ledger again
+   */
+  tryAgain = () => {
+    // clear error
+    this.setState({ errorMessage: null, waitAction: false }, () => {
+      this.trialNumber = 1;
+      ledger.getVersion();
+    });
+  }
+
+  render() {
+    const renderInitial  = () => {
+      return (
+        <div>
+          <p className="mt-4 mb-4">Do you want to connect to a hardware device (we currently support only Ledger), or want to start a software wallet?</p>
```

Comment:
I'd say we currently support only Ledger in another paragraph.

Reply:
Done